src/envs/pddl_env.py

```diff
@@ -0,0 +1,442 @@
+        problem_ast.accept(visitor)
+        pyperplan_problem = visitor.get_problem()
+        # Create the objects.
+        type_name_to_type = {t.name: t for t in self.types}
+        object_name_to_obj = {
+            o: Object(o, type_name_to_type[t.name])
+            for o, t in pyperplan_problem.objects.items()
+        }
+        objects = set(object_name_to_obj.values())
+        # Create the initial state.
+        predicate_name_to_predicate = {p.name: p for p in self.predicates}
+        init_ground_atoms = {
+            GroundAtom(predicate_name_to_predicate[a.name],
+                       [object_name_to_obj[n] for n, _ in a.signature])
+            for a in pyperplan_problem.initial_state
+        }
+        init = _ground_atoms_to_state(init_ground_atoms, objects)
+        # Create the goal.
+        goal = {
+            GroundAtom(predicate_name_to_predicate[a.name],
+                       [object_name_to_obj[n] for n, _ in a.signature])
+            for a in pyperplan_problem.goal
+        }
+        # Finalize the task.
+        task = Task(init, goal)
+        return task
+
+    def _strips_operator_to_parameterized_option(
+            self, op: STRIPSOperator) -> ParameterizedOption:
+        name = op.name
+        types = [p.type for p in op.parameters]
+        op_idx = self._ordered_strips_operators.index(op)
+        act_dims = self.action_space.shape[0]
+
+        def policy(s: State, m: Dict, o: Sequence[Object], p: Array) -> Action:
+            del m, p  # unused
+            ordered_objs = list(s)
+            obj_idxs = [ordered_objs.index(obj) for obj in o]
+            act_arr = np.zeros(act_dims, dtype=np.float32)
+            act_arr[0] = op_idx
+            act_arr[1:len(obj_idxs) + 1] = obj_idxs
+            return Action(act_arr)
+
+        def initiable(s: State, m: Dict, o: Sequence[Object],
+                      p: Array) -> bool:
+            del m, p  # unused
+            ground_atoms = _state_to_ground_atoms(s)
+            ground_op = op.ground(tuple(o))
+            return ground_op.preconditions.issubset(ground_atoms)
+
+        return utils.SingletonParameterizedOption(name,
+                                                  policy,
+                                                  types,
+                                                  initiable=initiable)
+
+
+class _FixedTasksPDDLEnv(_PDDLEnv):
+    """An env where tasks are induced by static PDDL problem files."""
+
+    @property
+    @abc.abstractmethod
+    def _pddl_problem_asset_dir(self) -> str:
+        """The name of the problem file directory in assets/pddl."""
+        raise NotImplementedError("Override me!")
+
+    @property
+    @abc.abstractmethod
+    def _train_problem_indices(self) -> List[int]:
+        """The problem file names should be task{idx}.pddl."""
+        raise NotImplementedError("Override me!")
+
+    @property
+    @abc.abstractmethod
+    def _test_problem_indices(self) -> List[int]:
```

Comment:
Return a list of indices corresponding to test problems. For each idx in the returned list, it is expected that there will be a problem file named task{idx}.pddl.